modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/Concurrent.kt

```diff
@@ -326,7 +326,10 @@ interface Concurrent<F> : Async<F> {
    * @see cancelableF for a version that can safely suspend impure callback registration code.
    */
   fun <A> cancelable(k: ((Either<Throwable, A>) -> Unit) -> CancelToken<F>): Kind<F, A> =
-    cancelableF { cb -> delay { k(cb) } }
+    cancelableF { cb ->
+      val token = k(cb)
```

Comment:
Was the encoding incorrect? Not saying I don't agree this encoding is better.

Reply:
No, it was incorrect before and it was causing a bunch of errors.

Reply:
Oh now I see. damn.